fix: Update ECS deployment guide to use Cloudformation (#18964)
cloudquery/cloudquery: website/pages/docs/deployment/ecs.md

cat cloudquery.yml | base64
```

> ## Step 2: Create Secret to CloudQuery API Key

Reviewer: ```suggestion
## Step 2: Create a Secret for the CloudQuery API Key
```

Reviewer 2: Btw in the previous section (sorry couldn't add a comment) there's no Windows version for the base64 encode. Apparently it's:

```
echo "your_string" | certutil -encodehex -f - 0
```

Author: This tutorial has been up for over a year and people have not asked about that. If we want to add windows support I think that can be done in a different PR

aws secretsmanager create-secret \
    --name CQ-APIKey \
    --description "API Key to authenticate with CloudQuery hub" \
    --secret-string "<CQ_API_KEY>"
```

> ## Step 3: Create CloudFormation template

Reviewer: ```suggestion
## Step 3: Create a CloudFormation template
```

    Type: String
  SecurityGroupIds:
    Description: Please enter a comma separated list of Security Group IDs that you want to attach to the Fargate node.
    Type: String   
  DestinationS3Bucket:
    Description: Please enter the name of the S3 bucket where you want to store the CloudQuery results
    Type: String
  CQConfiguration:
    Description: Please enter the CloudQuery configuration file encoded in base64
    Type: String
  CQApiKey:
    Description: ARN of the secret containing the CloudQuery API Key
    Type: String
  AWSMarketplace:
    Description: If you are using the AWS Marketplace version of CloudQuery, set this to true
    Type: String
    AllowedValues: [true, false]
    Default: false
Resources:
  #### ECS Cluster:
  ECSCluster:
    Type: 'AWS::ECS::Cluster'
  
  ScheduledWorkerTask:
    Type: 'AWS::ECS::TaskDefinition'
    Properties:
      RequiresCompatibilities:
        - FARGATE
      NetworkMode: awsvpc
      Cpu: 1024
      Memory: 2GB
      ExecutionRoleArn: !GetAtt ExecutionRole.Arn
      TaskRoleArn: !GetAtt ExecutionRole.Arn
      ContainerDefinitions:
        - Essential: 'true'
          Command: 

>             - "echo $CQ_CONFIG| base64 -d  > ./file.yml;/app/cloudquery sync ./file.yml --log-console --log-format json"

Reviewer 2: I think `base64 -d` is Linux only (`-D` on Mac I think); is this cloudformation template only applied on Linux instances?

Author: This is only Linux

          Image: !Sub ghcr.io/cloudquery/cloudquery:${CQVersion}
          Name: ScheduledWorker
          EntryPoint: 
            - "/bin/sh"
            - "-c"
          Environment:
            - Name: CQ_INSTALL_SRC
              Value: CLOUDFORMATION
            - Name: CQ_CONFIG
              Value: !Ref CQConfiguration
            - Name: CQ_AWS_MARKETPLACE_CONTAINER
              Value: !Ref AWSMarketplace
          Secrets:
               - ValueFrom: !Ref CQApiKey
                 Name: CLOUDQUERY_API_KEY
          LogConfiguration:
            LogDriver: awslogs
            Options:
              awslogs-group: !Ref LogGroup
              awslogs-region: !Ref AWS::Region
              awslogs-stream-prefix: !Ref AWS::StackName
  LogGroup:
    DeletionPolicy: Retain
    UpdateReplacePolicy: Retain  
    Type: AWS::Logs::LogGroup
    Properties: 
      RetentionInDays: 14
  # Scheduler Configurations
  Schedule:
    Type: AWS::Scheduler::Schedule
    Properties: 
      FlexibleTimeWindow: 
        Mode: "OFF"
      ScheduleExpression: !Ref ScheduleExpression
      State: ENABLED
      Target: 

>           Arn: !GetAtt ECSCluster.Arn

Reviewer 2: I think this is a problem here with the inconsistent spacing. It seems to be 4 spaces here and 2 spaces mostly elsewhere.

Author: fixed in [128340e](https://github.com/cloudquery/cloudquery/pull/18964/commits/128340e7b2333502de16087e63b21d154ed4c2bc)

        - PolicyName: DenyData
          PolicyDocument:
            Version: 2012-10-17
            Statement:
              - Effect: Deny
                NotResource: 
                  - !Sub arn:${AWS::Partition}:s3:::${DestinationS3Bucket}/*
                Action:
                  - s3:GetObject
              -
                Effect: Deny
                Resource: '*'
                Action:
                  - cloudformation:GetTemplate
                  - dynamodb:GetItem
                  - dynamodb:BatchGetItem
                  - dynamodb:Query
                  - dynamodb:Scan
                  - ec2:GetConsoleOutput
                  - ec2:GetConsoleScreenshot
                  - kinesis:Get*
                  - lambda:GetFunction
                  - logs:GetLogEvents
                  - sdb:Select*
                  - sqs:ReceiveMessage
      ManagedPolicyArns:
        - 'arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy'
        - 'arn:aws:iam::aws:policy/ReadOnlyAccess'
        - 'arn:aws:iam::aws:policy/AWSMarketplaceMeteringFullAccess'
Outputs:
  ClusterId:
    Value: !Ref ECSCluster
  TaskArn:
    Value: !Ref ScheduledWorkerTask
```

> ## Step 3: Deploy the CloudFormation Template

Reviewer 2: Step 3 is duplicated here

Author: Fixed in [caeb720](https://github.com/cloudquery/cloudquery/pull/18964/commits/caeb720af516f894e04dd7b5746a7ded81b81bed)

You can deploy the CloudFormation template using the `aws cloudformation deploy` command. This command will create the required resources for the deployment of CloudQuery on AWS ECS with Fargate. If you are using the AWS Marketplace version of CloudQuery, you can set the `AWSMarketplace` parameter to `true`.

``` bash
aws cloudformation deploy --template-file cloudquery-ecs-resources.yaml --stack-name <STACK_NAME> --parameter-overrides CQApiKey=<SECRET_ARN> CQVersion=latest PrivateSubnetIds=<SUBNET_ID_1>,<SUBNET_ID_2> SecurityGroupIds=<SecurityGroup_ID_1>  DestinationS3Bucket=<DESTINATION_BUCKET> CQConfiguration=<BASE64_ENCODED_CONFIG>
```

> ## Step 4: Run a test CloudQuery sync

Reviewer: ```suggestion
## Step 4: Run a CloudQuery sync
```
